index.js

```diff
@@ -0,0 +1,30 @@
+const defaultMap = require('./default_map');
+
+const coreFields = [
+  'v',
+  'level',
+  'name',
+  'hostname',
+  'pid',
+  'time',
+  'msg',
+  'src'
+];
+
+module.exports = function bunyanEncoder(mapCoreFields = defaultMap, stream = process.stdout) {
+  return {
+    write: function (rec) {
+      if (typeof(rec) === 'string' || (rec instanceof String)) {
+        throw new Error('Stream must be raw');
+      }
+
+      const recCopy = JSON.parse(JSON.stringify(rec));
```

Comment:
Agree that performance suffers. To me, it is a bad practice to mutate an object that came as a parameter. The caller may not expect this and it could lead to bugs. For this particular case we could neglect this for performance sake since the `rec` object is not used anywhere in bunyan after `write` method is called.